device-registry/src/main/scala/org/genivi/sota/device_registry/JsonComparison.scala

```diff
@@ -0,0 +1,70 @@
+package org.genivi.sota.device_registry
+
+import io.circe.{Json, JsonObject}
+
+import scala.collection.mutable
+
+object JsonComparison {
+
+  def compareJsonObjects(json1: JsonObject, json2: JsonObject): Json = {
+    val commonJson = mutable.MutableList.empty[(String, Json)]
```

Comment:
this can be done with a immutable val using foldLeft, `json1.fields.foldLeft` and accumulating the values on `json2`.

Otherwise it's fine, we could probably do this with a traverse with circe but we shouldn't complicate this too much.